drivers/sensor/bh1749/bh1749.c

```diff
@@ -0,0 +1,234 @@
+				   return -EIO;
+			   }
+
+	if (IS_ENABLED(CONFIG_BH1749_TRIGGER)) {
+		/* Clearing interrupt by reading INTERRUPT register */
+		u8_t interrupt_disabled;
+		if (i2c_reg_read_byte(data->i2c, DT_ROHM_BH1749_0_BASE_ADDRESS,
+				      BH1749_INTERRUPT, &interrupt_disabled)) {
+			LOG_ERR("Could not disable sensor interrupt.");
+			return -EIO;
+		}
+
+		if (gpio_pin_enable_callback(data->gpio,
+					     DT_ROHM_BH1749_0_INT_GPIOS_PIN)) {
+			LOG_ERR("Could not enable pin callback");
+			return -EIO;
+		}
+	}
+
+	return 0;
+}
+
+static int bh1749_channel_get(struct device *dev,
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	struct bh1749_data *data = dev->driver_data;
+
+	switch (chan) {
+	case SENSOR_CHAN_RED:
+		val->val1 = sys_le16_to_cpu(
+			data->sample_rgb_ir[BH1749_SAMPLE_POS_RED]);
+		val->val2 = 0;
+		break;
+	case SENSOR_CHAN_GREEN:
+		val->val1 = sys_le16_to_cpu(
+			data->sample_rgb_ir[BH1749_SAMPLE_POS_GREEN]);
+		val->val2 = 0;
+		break;
+	case SENSOR_CHAN_BLUE:
+		val->val1 = sys_le16_to_cpu(
+			data->sample_rgb_ir[BH1749_SAMPLE_POS_BLUE]);
+		val->val2 = 0;
+		break;
+	case SENSOR_CHAN_IR:
+		val->val1 = sys_le16_to_cpu(
+			data->sample_rgb_ir[BH1749_SAMPLE_POS_IR]);
+		val->val2 = 0;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int bh1749_check(struct device *dev)
+{
+	struct bh1749_data *data = dev->driver_data;
+
+	u8_t manufacturer_id;
+
+	if (i2c_reg_read_byte(data->i2c, DT_ROHM_BH1749_0_BASE_ADDRESS,
+			      BH1749_MANUFACTURER_ID, &manufacturer_id)) {
+		LOG_ERR("Failed when reading manufacturer ID");
+		return -EIO;
+	}
+
+	LOG_DBG("Manufacturer ID: 0x%02x", manufacturer_id);
+
+	if (manufacturer_id != BH1749_MANUFACTURER_ID_DEFAULT) {
+		LOG_ERR("Invalid manufacturer ID: 0x%02x", manufacturer_id);
+		return -EIO;
+	}
+
+	u8_t part_id;
+
+	if (i2c_reg_read_byte(data->i2c, DT_ROHM_BH1749_0_BASE_ADDRESS,
+			      BH1749_SYSTEM_CONTROL, &part_id)) {
+		LOG_ERR("Failed when reading part ID");
+		return -EIO;
+	}
+
+	if ((part_id & BH1749_SYSTEM_CONTROL_PART_ID_Msk) !=
+	    BH1749_SYSTEM_CONTROL_PART_ID) {
+		LOG_ERR("Invalid part ID: 0x%02x", part_id);
+		return -EIO;
+	}
+
+	LOG_DBG("Part ID: 0x%02x", part_id);
+
+	return 0;
+}
+
+static int bh1749_sw_reset(struct device *dev)
+{
+	return i2c_reg_update_byte(dev, DT_ROHM_BH1749_0_BASE_ADDRESS,
+				   BH1749_SYSTEM_CONTROL,
+				   BH1749_SYSTEM_CONTROL_SW_RESET_Msk,
+				   BH1749_SYSTEM_CONTROL_SW_RESET);
+}
+
+static int bh1749_rgb_measurement_enable(struct device *dev, bool enable)
+{
+	u8_t en = enable ?
+		  BH1749_MODE_CONTROL2_RGB_EN_ENABLE :
+		  BH1749_MODE_CONTROL2_RGB_EN_DISABLE;
+
+	return i2c_reg_update_byte(dev, DT_ROHM_BH1749_0_BASE_ADDRESS,
+				   BH1749_MODE_CONTROL2,
+				   BH1749_MODE_CONTROL2_RGB_EN_Msk,
+				   en);
+}
+
+static int bh1749_init(struct device *dev)
+{
+	int err;
+	struct bh1749_data *data = dev->driver_data;
+
+	/* Activation time 2ms */
+	k_sleep(2);
```

Comment:
You should not do it this way as this will delay the initialization of the whole system. Instead, you should use `k_delayed_work_submit` to defer the remaining initialization of this sensor after binding to the I2C bus here.

See how it is done in the [PAW3212](https://github.com/NordicPlayground/fw-nrfconnect-nrf/blob/5af18a03d946c025aa8b2e2e29186d138acbb35a/drivers/sensor/paw3212/paw3212.c#L457), for instance.